lib/rester/middleware/correlation_id.rb

```diff
@@ -0,0 +1,16 @@
+module Rester
+  module Middleware
+    ##
+    # Sets the correlation id for the current thread if one is passed in the
+    # request header. Clean up the correlation id for this thread before
+    # responding.
+    class CorrelationId < Base
+      def call(env)
+        Rester.correlation_id = env['X-Rester-Correlation-ID']
+        super.tap { |response|
+          Rester.correlation_id = nil
```

Comment:
I don't think this will work behind Rack, which will automatically rename the header to `HTTP_X_RESTER_CORRELATION_ID`.